ndcube/conftest.py

```diff
@@ -318,6 +456,57 @@ def extra_coords_sharing_axis():
 # NDCube Fixtures
 ################################################################################
 
+@pytest.fixture
+def ndcube_gwcs_4d_ln_lt_l_t(gwcs_4d_t_l_lt_ln):
+    shape = (5, 8, 10, 12)
+    gwcs_4d_t_l_lt_ln.array_shape = shape
+    data_cube = data_nd(shape)
+    return NDCube(data_cube, wcs=gwcs_4d_t_l_lt_ln)
+
+@pytest.fixture
+def ndcube_gwcs_3d_ln_lt_l(gwcs_3d_lt_ln_l):
+    shape = (2, 3, 4)
+    gwcs_3d_lt_ln_l.array_shape = shape
+    data_cube = data_nd(shape)
+    return NDCube(data_cube, wcs=gwcs_3d_lt_ln_l)
+
+@pytest.fixture
+def ndcube_gwcs_3d_rotated(gwcs_3d_lt_ln_l, simple_extra_coords_3d):
+    data_rotated = np.array([[[1, 2, 3, 4, 6], [2, 4, 5, 3, 1], [0, -1, 2, 4, 2], [3, 5, 1, 2, 0]],
+                             [[2, 4, 5, 1, 3], [1, 5, 2, 2, 4], [2, 3, 4, 0, 5], [0, 1, 2, 3, 4]]])
+    cube = NDCube(
+        data_rotated,
+        wcs=gwcs_3d_lt_ln_l)
+    cube._extra_coords = simple_extra_coords_3d
+    return cube
+
+@pytest.fixture
+def ndcube_gwcs_3d_ln_lt_l_ec_dropped_dim(gwcs_3d_lt_ln_l, time_and_simple_extra_coords_2d):
+    shape = (2, 3, 4)
+    gwcs_3d_lt_ln_l.array_shape = shape
+    data_cube = data_nd(shape)
+    cube =  NDCube(data_cube, wcs=gwcs_3d_lt_ln_l)
+    cube._extra_coords = time_and_simple_extra_coords_2d[0]
+    return cube
+
+@pytest.fixture
+def ndcube_gwcs_3d_ln_lt_l_ec_q_t_gc(gwcs_3d_lt_ln_l):
+    shape = (3, 3, 4)
+    gwcs_3d_lt_ln_l.array_shape = shape
+    data_cube = data_nd(shape)
+    cube =  NDCube(data_cube, wcs=gwcs_3d_lt_ln_l)
+    coord1 = 1 * u.m
+    cube.global_coords.add('name1', 'custom:physical_type1', coord1)
+    cube.extra_coords.add("time", 0, time_lut(shape))
+    cube.extra_coords.add("exposure_lut", 1, range(shape[1]) * u.s)
```

Comment:
I have not added the extra_coords with `SkyCoordTableCoordinate` because there seems to be unexpected behavior when using `assert_extra_coords_equal` to verify the roundtrip serialization.